### PR TITLE
Update and additions to pytorch-cuda12 testing instructions

### LINKED ISSUE #2328: [BUG] Documentation on `pytorch-cuda12` Missing Beginning of `gcloud` Command
### Which image/versions are related to this issue/feature request?

pytorch-cuda12

### Issue/Feature description

In the [pytorch-cuda12](https://github.com/chainguard-images/images/blob/main/images/pytorch-cuda12/TESTING.md#installation) documentation on testing, the first step is creating a VM with GPUs.

It is currently written as below:

```
--zone=us-central1-a \
--machine-type=n1-standard-4 \
--accelerator="type=nvidia-tesla-t4,count=2" \
--boot-disk-size=200GB \
--boot-disk-type=pd-standard \
--image-family=ubuntu-2204-lts \
--image-project=ubuntu-os-cloud \
--maintenance-policy=TERMINATE \
--restart-on-failure
```

I don't know much about GCP, but I suspect that the command should be more like:

```
gcloud compute instances create INSTANCE_NAME \
--zone=us-central1-a \
--machine-type=n1-standard-4 \
--accelerator="type=nvidia-tesla-t4,count=2" \
--boot-disk-size=200GB \
--boot-disk-type=pd-standard \
--image-family=ubuntu-2204-lts \
--image-project=ubuntu-os-cloud \
--maintenance-policy=TERMINATE \
--restart-on-failure
```

Ignore me if I'm confused and close this issue :)

@EyeCantCU

### PR DESCRIPTION
**Pull request by smythp**
Fixed and updated instance creation command

Added step-by-step instructions

Added more test commands

Added links to resources

Resolves #2328

This is a documentation-only update.

### CODE REVIEW

File: images/pytorch-cuda12/TESTING.md [hunk 1]
```diff
@@ -1,77 +1,97 @@
...
+Install [Docker Engine](https://docs.docker.com/engine/install/ubuntu/).
 
-https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#ubuntu
+Run the image:
 
+```
+docker run --rm -i -t \
+    --gpus all \
+    cgr.dev/chainguard/pytorch:latest
+```
```
Comment (srishtih) [inline images/pytorch-cuda12/TESTING.md:91]:
```
docker run --rm -i -t \
    --gpus all \
    cgr.dev/chainguard/pytorch-cuda12:latest
```

File: images/pytorch-cuda12/TESTING.md [hunk 2]
```diff
@@ -1,77 +1,97 @@
...
-    ```
+    ```bash
+    gcloud compute instances create $INSTANCE_NAME \
+      --zone=$ZONE \
+      --image-family=$IMAGE_FAMILY \
+      --image-project=deeplearning-platform-release \
+      --maintenance-policy=TERMINATE \
+      --accelerator="type=nvidia-tesla-v100,count=1" \
+      --metadata="install-nvidia-driver=True"
+  ```
```
Comment (srishtih) [inline images/pytorch-cuda12/TESTING.md:44]: Minor formatting
    ```bash
    gcloud compute instances create $INSTANCE_NAME \
      --zone=$ZONE \
      --image-family=$IMAGE_FAMILY \
      --image-project=deeplearning-platform-release \
      --maintenance-policy=TERMINATE \
      --accelerator="type=nvidia-tesla-v100,count=1" \
      --metadata="install-nvidia-driver=True"
    ```

File: images/pytorch-cuda12/TESTING.md [hunk 3]
```diff
@@ -1,77 +1,97 @@
...
 
+```
+docker run --rm -i -t \
+    --gpus all \
+    cgr.dev/chainguard/pytorch:latest
+```
 
+## Documentation and Resources
 
+- [ Deep Learning VM  Docs](https://cloud.google.com/deep-learning-vm/docs?_gl=1*2hwanx*_ga*MjUwMTMwMjA5LjE3MTE5OTYxNTU.*_ga_WH2QY8WWF5*MTcxMjg0MzE4OS40LjEuMTcxMjg0NzI3NS4wLjAuMA..&_ga=2.149113461.-250130209.1711996155)
```
Comment (srishtih) [inline images/pytorch-cuda12/TESTING.md:95]: [Deep Learning VM  Docs](https://cloud.google.com/deep-learning-vm/docs)

**Review by srishtih — DISMISSED** [review]

File: images/pytorch-cuda12/TESTING.md [hunk 4]
```diff
@@ -1,77 +1,97 @@
...
+
+1. If you haven't already, install the [gcloud CLI](https://cloud.google.com/sdk/docs/install).
+2. In GCP, create a project and grab the project ID or find the ID of
+   your existing project. The ID is a descriptive string, not a
+   number.
+3. Run `gcloud init` in the terminal to authenticate to GCP. You may
+   want to add the ID above at this step. You can set the project
+   later with `gcloud config set project <project-id>`.
+4. Once you're authenticated, review and set these environment
+   variables as below. . GPU aren't available in all zones and this
```
Comment (jspeed-meyers) [inline images/pytorch-cuda12/TESTING.md:19]:
```suggestion
   variables as below. GPU aren't available in all zones and this
```

**Review by jspeed-meyers — APPROVED** [review]
LGTM except a tiny typo. Nice work, @smythp!

**Comment by jspeed-meyers** [comment]
BTW, I don't have merge rights :) Someone else will need to merge it. Feel free to drop this in the `#want-to-buy` channel if you haven't already.

**Review by srishtih — APPROVED** [review]

**Pull request merged**